plutus-ledger-api/src/PlutusLedgerApi/Common/Eval.hs

```diff
@@ -106,33 +106,71 @@ mkTermToEvaluate ll pv script args = do
     through (liftEither . first DeBruijnError . UPLC.checkScope) appliedT
 
 toMachineParameters :: MajorProtocolVersion -> EvaluationContext -> DefaultMachineParameters
-toMachineParameters _ = machineParameters
+toMachineParameters pv (EvaluationContext lv toSemVar machParsList) =
+    case lookup (toSemVar pv) machParsList of
+        Nothing -> error $ Prelude.concat
+            ["Internal error: ", show lv, " does not support protocol version ", show pv]
+        Just machPars -> machPars
 
 {-| An opaque type that contains all the static parameters that the evaluator needs to evaluate a
-script.  This is so that they can be computed once and cached, rather than being recomputed on every
+script. This is so that they can be computed once and cached, rather than being recomputed on every
 evaluation.
+
+Different protocol versions may require different bundles of machine parameters, which allows us for
```

Comment:
Maybe make this into a Note?  It seems  quite important.  It would seem to have some bearing on [this](https://github.com/IntersectMBO/plutus/pull/5851/files#r1565862928), for example.

Reply:
> Maybe make this into a Note?

I kinda want this to be the Haddock for extra transparency. I do reference the Haddock of a function instead of a Note sometimes.